effekt/shared/src/main/scala/effekt/core/PatternMatchingCompiler.scala

```diff
@@ -179,12 +179,20 @@ object PatternMatchingCompiler {
 
       // used to make up new scrutinees
       val varsFor = mutable.Map.empty[Id, List[ValueVar]]
-      def fieldVarsFor(constructor: Id, fieldTypes: List[ValueType]): List[ValueVar] =
-        varsFor.getOrElseUpdate(constructor, fieldTypes.map { tpe => ValueVar(TmpValue("y"), tpe) })
+      def fieldVarsFor(constructor: Id, fieldPatternsAndTypes: List[(Pattern, ValueType)]): List[ValueVar] =
+        varsFor.getOrElseUpdate(
+          constructor,
+          fieldPatternsAndTypes.map { case (pat, tpe) =>
+            val idHint = pat match
+              case Pattern.Any(id) => id.name.name // if we have a pattern name, use it as a hint!
+              case _ => "y"
+            ValueVar(TmpValue(idHint), tpe)
```

Comment:
Here I could also use the previous `id` directly if we have one, i.e.:
 ```suggestion
          fieldPatternsAndTypes.map { case (pat, tpe) =>
            val patId = pat match
              case Pattern.Any(id) => id // if we have a pattern id, use it!
              case _ => TmpValue("y")
            ValueVar(patId, tpe)
```

The downside is that then variables are not globally unique, only locally. :(

Is there some other option?